lib/authentication/view/edit_profile_page.dart

```diff
@@ -129,9 +133,12 @@ class _EditProfilePageState extends State<EditProfilePage> {
           color: Theme.of(context).accentColor,
           width: 130,
           onTap: () async {
+            correctPassword = false;
+            if (await authProvider.verifyPassword(oldPasswordController.text)) {
```

Comment:
Having a variable for this is kinda hacky. Why not just move this verification call in the validator directly?

(note that the validator(s) are called when we call `changePasswordKey.currentState.validate()`, that's how `Form`s are used)

Reply:
Well, it's an async function so I can't call it inside the validator of the FormField. And the validator needs to know if the password is right or not. Can I pass the information without using a variable?

Reply:
Ugh, okay. Let's leave it like this then. But the code itself can be improved, you can just do `if (!correctPassword)` above. And here you can just do `correctPassword = await ...`, no need for the `if`.